CR1: add styler.py script for review

j-faria opened the pull request:
This script provides a `styler` decorator that helps in making publication-ready figures, with the correct figure and font sizes.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/iastro-pt/codereviewclub/1)
<!-- Reviewable:end -->

j-faria committed:
CR1: add styler.py script for review

DanielAndreasen commented on styler.py:
```diff
@@ -0,0 +1,218 @@
+#### styler should also provide colors for consistency!
+## are these ok?
+## should actually be a dictionary with sane keys...
+colors = ['#4878cf', '#6acc65', '#d65f5f', '#b47cc7', '#c4ad66', '#77bedb']
```
Atom shows better colours, but maybe add some keys here?
`colors = {'blue': '#4878cf', 'green': ...}`

DanielAndreasen commented on styler.py:
```diff
@@ -0,0 +1,218 @@
+    """
+    This function deals with the figure and axes
+    and makes them suitable to plot in the thesis
+    """
```
Which keywords can be used?

An example in the documentation would also be very helpful

DanielAndreasen commented on styler.py:
```diff
@@ -0,0 +1,218 @@
+            # where the minus sign is replaced by the short hyphen
+            rv = re.sub(r'-', r'\mhyphen', rv)
+
+        rv = rv.replace('.0', '')
```
This might remove unwanted `.0`.

    if rv.endswith('.0'):
        rv = rv.replace('.0', '')

or something similar

DanielAndreasen reviewed (commented):
Suggestions from CRC

j-faria commented on styler.py:
```diff
@@ -0,0 +1,218 @@
+# matplotlib parameters for publication plots
+import matplotlib.pyplot as plt
+import matplotlib.ticker as ticker
+from matplotlib import gridspec
```
:+1:

sousasag commented on styler.py:
```diff
@@ -0,0 +1,218 @@
+full_width = 6.5046152778 # in inches = \linewidth (in fullwidth environment) = 468.3323pt
+
+### for the A&A article class
+AandA_width = 3.543311946  # in inches = \hsize = 256.0748pt
```
AandA width is 3.543311945991

DanielAndreasen replied:
How did you get this information?

j-faria replied:
Put `\the\hsize`in the Latex inside a figure and it prints 256.0748pt, then convert to inches

jason-neal commented:
So is this all complete? 
Does this get merged now or should it stay untill the next one?

j-faria committed:
Merge branch 'master' of https://github.com/iastro-pt/CodeReviewClub into CR1

j-faria committed:
implement corrections from review; add some documentation; put in subfolder

j-faria committed:
add readme file to CR1

j-faria committed:
Update README.md

j-faria commented:
I changed the script to a `CR1` folder and added a small readme file, as suggested in #2.
Corrected the points raised during the CRC meeting and added some documentation to the styler decorator. Ready to merge?

DanielAndreasen commented:
Seems fine to me 👍
